### ISSUE TITLE
[Modal] Modal with JS-derived width should be properly centered with modal refresh

### ISSUE THREAD

**Issue by davidmccoy** [Semantic-Org/Semantic-UI#4159]
**tl;dr:** After setting the width of a modal with JavaScript, further window resizes/modal refreshes incorrectly center the modal based on the CSS-derived width.

I'm working on an image-focused app that uses modals to display images à la Instagram. However, modals often become too long to fit on the screen when displaying portrait-oriented images. I would like every modal to fit on the screen so when .modal('can fit') returns false I resize the modal with a custom width that ensures the resulting modal will fit on the screen.

Unfortunately, this results in some unexpected behavior. The modal fits the screen and is properly centered when its width is first set by my JS; however, any other window resize/modal refresh events cause the modal become uncentered. It appears that the modal is being "centered" based on it's original CSS-derived width rather than it's new/current JS-derived width.

Is there a better way to ensure that a modal always fits on the screen? Or am I missing something?

Thanks for your help!
